packages/asset-service/src/service/AssetService.ts

```diff
@@ -12,15 +14,35 @@ export const flattenAssetData = (assetData: BaseAsset[]): Asset[] => {
     flatAssetData.push(newAsset)
     if (baseAsset.tokens) {
       for (const tokenAsset of baseAsset.tokens) {
-        flatAssetData.push({
+        const flatAssetDatum = {
           ...tokenAsset,
           chain: baseAsset.chain,
           network: baseAsset.network,
           slip44: baseAsset.slip44,
           explorer: baseAsset.explorer,
           explorerAddressLink: baseAsset.explorerAddressLink,
           explorerTxLink: baseAsset.explorerTxLink
-        })
+        }
+        if (!flatAssetDatum.icon) {
+          const { tokenId } = caip19.fromCAIP19(flatAssetDatum.caip19)
+          const options: IdenticonOptions = {
+            identiconImage: {
+              size: 128,
+              background: [45, 55, 72, 255]
+            },
+            identiconText: {
+              symbolScale: 7,
+              enableShadow: true
+            }
+          }
+          if (tokenId)
+            flatAssetDatum.icon = getRenderedIdenticonBase64(
```

Comment:
We'll want to generate icons for base assets as well as tokens. Currently, there aren't a lot of base assets, but in the future there may be some that may not have an icon. So I think this same logic applied to the base assets as well should work.